lib/Basics/UnshackledMutex.cpp

```diff
@@ -25,28 +25,29 @@
 
 #include "Basics/debugging.h"
 
-void arangodb::basics::UnshackledMutex::lock() {
-  auto guard = std::unique_lock(_mutex);
-  _cv.wait(guard, [&] { return !_locked; });
-  TRI_ASSERT(!_locked);
+namespace arangodb::basics {
+
+void UnshackledMutex::lock() noexcept {
+  absl::MutexLock guard{
+      &_mutex,
+      absl::Condition{+[](bool const* locked) noexcept { return !*locked; },
+                      &std::as_const(_locked)}};
```

Comment:
What is this `+` doing here? `absl::Condition{  +  [](bool const* locked)`

Reply:
Needs to get function pointer instead of lambda type

Reply:
absl::Condition have explicit ctor if I remember correctly

Reply:
Hmm, no. I think it needed for template deduction
https://github.com/abseil/abseil-cpp/blob/master/absl/synchronization/mutex.h#L671

Reply:
>Needs to get function pointer instead of lambda type

Its valid C++ way to get it
https://stackoverflow.com/questions/18889028/a-positive-lambda-what-sorcery-is-this

Reply:
Thanks, I have never seen this, but I think I could have used this already on occasion. :)